src/hip_memory.cpp

```diff
@@ -1266,8 +1266,12 @@ hipError_t hipMemcpyToSymbol(void* dst, const void* src, size_t count,
 
     tprintf(DB_MEM, " symbol '%s' resolved to address:%p\n", symbol_name, dst);
 
-    return ihipLogStatus(
-        hipMemcpy(static_cast<char*>(dst) + offset, src, count, kind));
+    if (dst == nullptr) {
+        return ihipLogStatus(hipErrorInvalidSymbol);
```

Comment:
I feel hipErrorInvalidValue is more appropriate than hipErrorInvalidSymbol.

Reply:
The inline implementation @ include/hip/hcc_detail/hip_runtime_api.h:1594 takes in the symbolName and determines the dst pointer using hipGetSymbolAddress. So if dst is null it means the symbolName was probably invalid. Hence returning hipErrorInvalidSymbol is the right thing which is also what the CUDA equivalent API would be returning.